Makefile

```diff
@@ -17,11 +17,16 @@ TOOLS += agent-ctl
 
 STANDARD_TARGETS = build check clean install test vendor
 
+default: all
+
+all: logging-crate-tests build
```

Comment:
Does `logging-crate-tests` will run twice? `build` target in sub-directories will also trigger `logging-crate-tests` too.

Reply:
@liubin - it might yes. The problem is we need to ensure the logging crate tests run however the build is invoked - either running `make` from the top-level, or running `make -C src/agent` for example. I'm not sure how to avoid running it multiple times whilst also ensuring that it _does_ run. Any ideas?

ftr, running the logging crate tests twice takes < 300ms on a laptop system so it's potentially unecessary to run multiple times, but it isn't going to increase the build time much.

Ideally, we want `cargo test` and `cargo test --release` to run when any of the other components run `cargo build`, but I don't think that's possible. I did think of one way, but it's horrid: create `pkg/logging/build.rs` which runs `make`! ;-)

Reply:
It's a hack, but I could then remove all the changes to the other makefiles.